### ISSUE TITLE
Tennistv.com streaming 

### ISSUE THREAD

**Issue by LDR1987** [streamlink/streamlink#2666]
Tennistv.com streaming 

Hi,

I have paid account and would like to run tennistv.com streams through streamlink as they browser solution doesnt work for me anymore and started to freeze every 10 seconds. 

Example stream URLs

https://vod-i-tennistv.akamaized.net/vod/42b1d809-b487-4828-a050-77c1e6fa0956/42b1d809-b487-4828-a050-77c1e6fa0956.ism/QualityLevels(1800000)/Manifest(video,format=m3u8-aapl-v3,audiotrack=audio,filter=trimmed2)

How can I input this one in streamlink in order to get it stream in VLC for example?

Thank You

**Comment by bastimeyer** [comment]
See https://streamlink.github.io/cli.html#playing-built-in-streaming-protocols-directly

Since you are trying to watch an HLS stream, you have to set `hls://` in front of the URL or replace `https://` with `hls://`.

And in case that the URL includes special characters which your command line shell might interpret differently, you have to put the URL in (single-)quotes.

**Comment by LDR1987** [comment]
> See https://streamlink.github.io/cli.html#playing-built-in-streaming-protocols-directly
> 
> Since you are trying to watch an HLS stream, you have to set `hls://` in front of the URL or replace `https://` with `hls://`.
> 
> And in case that the URL includes special characters which your command line shell might interpret differently, you have to put the URL in (single-)quotes.

OK, now I get:
hls://vod-i-tennistv.akamaized.net/vod/42b1d809-b487-4828-a050-77c1e6fa0956/42b1d809-b487-4828-a050-77c1e6fa0956.ism/QualityLevels(1200000)/Manifest(video,format=m3u8-aapl-v3,audiotrack=audio,filter=trimmed2)
Available streams: live (worst, best)

how can I dig available qualities and run it??

**Comment by bastimeyer** [comment]
https://streamlink.github.io/cli.html#tutorial

**Comment by LDR1987** [comment]
> https://streamlink.github.io/cli.html#tutorial

Well, I am probably stupid but the way I got it streamlink should run one of the available streams after "Available streams:live" line is generated but it is not in my case. Also, I have no idea how you can use streamlink config file. I opened it in Notepad and there is bunch of possible config lines but where I should put them if I want to use any of them? Like path to players for example??

**Comment by LDR1987** [comment]
> > https://streamlink.github.io/cli.html#tutorial
> 
> Well, I am probably stupid but the way I got it streamlink should run one of the available streams after "Available streams:live" line is generated but it is not in my case. Also, I have no idea how you can use streamlink config file. I opened it in Notepad and there is bunch of possible config lines but where I should put them if I want to use any of them? Like path to players for example??

For example there is:
# VLC
#player="C:\Program Files (x86)\VideoLAN\VLC\vlc.exe"

where I put this player argument in order for srtreamlink automatically run VLC??

**Comment by LDR1987** [comment]
Finally!! I got it.  Thank You bastimeyer you have pointed me to right direction :)

**Comment by gray-morley** [comment]
@LDR1987 are you still able to do this? I get an 'invalid URL' message when I try. Curious if were able to find a different way of getting the original stream (I have a paid tennistv subscription)

**Comment by LDR1987** [comment]
Hi,

no, I dont even remember where I paid and watch tennis.tv last time, werent
watching atp tennis for long time now.

In the past I did download mainly because my internet connection was very
slow, but then it got much better and I was able to stream tennis.tv
directly so no need to first download and only then watch.

But really with big 3 collapsing and now only struggling Djokovic left my
entusiasm for watching tennis dicreased dramatically in recent years, I
rarely check scores these days.
